### ISSUE TITLE
Provide ability to optionally surface semi-structured data from properties service

### ISSUE THREAD

**Issue by cbryanreapit** [reapit/foundations#3890]
**Background context**
Our customers have historically stored custom fields bespoke to their business or region using semi-structured storage. This information has allowed our customers to evolve the data they work with without requiring the software to be updated. Generally, textual data, date stamps and boolean content is available here. We envisage this only be used by customers who have in house development resource or developers working solely on their behalf.

**Specification**
- Allow semi-structured data (extras) to be surfaced 
- Read only for both GET and GET by id methods, no writing of this data 
- Only for property service in first instance
- Mechanism to be driven by query string - developers need to include one or more query string corresponding to the field they want to grab
- Eg. `GET /properties?extrasField=pFieldName1&extrasField=pFieldName2`
- Include new attribute in property payload 'extrasField' that becomes populated when query string provided and data available
- Availability of fields to be controlled by a whitelist held with the customers configuration information
- Developers should submit a standard GitHub issue to request access to certain fields, which platform team can review vs certain criteria
- This process will need to be documented in the developer portal appropriately 
- We may want to whitelist a field if already exposed, sensitive or planned to be a structured field later
- If a request is made for a non approved field, return with a 400 (TBC)
- Implement where possible in a futureproof way, requiring no future updates to the identity package
- 
**Notes**
- Case insensitive on whitelist and query strings
- See CoreAPI project for early implementation of this
- Be aware of UTC timezone conversion (though timestamps aren't likely to be there, just a note until we know otherwise)
- Identity package requires an update
- Application layer is the right place for this logic

**Comment by david-silverink** [comment]
Could you confirm how to find the actual field names to be used in the query string? E.g. apparently BER fields are now available to access in the payload, but as they are non-standard fields, the actual field names for BER number, code and EPI need to be determined before we can add them to the query.

**Comment by cbryanreapit** [comment]
Hi @david-silverink - the process is outlined in our developer documentation, but you just need to submit a new ticket to this board using the 'Extras field request' template that you'll be presented with. This will privude our team with all of the information we need to turn this on for you.

![image](https://user-images.githubusercontent.com/34191069/115703200-be03b000-a361-11eb-8c9c-20ed4463958d.png)
